src/encoder/text.rs

```diff
@@ -45,88 +47,114 @@ impl Encoder for TextEncoder {
             }
 
             // Write `# TYPE` header.
-            let metric_type = mf.get_field_type();
+            let metric_type = mf
+                .r#type
+                .map(MetricType::from_i32)
+                .flatten()
+                .unwrap_or_default();
             let lowercase_type = format!("{:?}", metric_type).to_lowercase();
             writer.write_all(b"# TYPE ")?;
             writer.write_all(name.as_bytes())?;
             writer.write_all(b" ")?;
             writer.write_all(lowercase_type.as_bytes())?;
             writer.write_all(b"\n")?;
 
-            for m in mf.get_metric() {
+            for m in &mf.metric {
                 match metric_type {
-                    MetricType::COUNTER => {
-                        write_sample(writer, name, None, m, None, m.get_counter().get_value())?;
+                    MetricType::Counter => {
+                        let value = match &m.counter {
+                            Some(v) => v.value.unwrap_or_default(),
+                            None => 0.0,
```

Comment:
If metric type is `Counter`, but `m.counter` is `None`, I would expect an error to be thrown instead of silently using `0.0`. Am I missing something?

Reply:
Simply migration too:
```rust
    pub fn get_counter(&self) -> &Counter {
        self.counter.as_ref().unwrap_or_else(|| Counter::default_instance())
    }
```

Also, I think it's a good choice to return an error.